src/querydb/store.rs

```diff
@@ -0,0 +1,220 @@
+        let mut chain_tip_table: Table<ChainTipKeyType, ChainTipValueType> = write_tx
+            .open_table(CHAIN_TIP_TABLE)
+            .map_err(|e| StoreError::ReDBError(ReDBError::TableError(e)))?;
+        let _ = chain_tip_table.insert(
+            chain_tip_table
+                .len()
+                .map_err(|e| StoreError::ReDBError(ReDBError::InsertError(e)))?
+                + 1,
+            block.hash().deref(),
+        );
+        Ok(block)
+    }
+
+    fn store_txs(write_tx: &WriteTransaction, block: &MultiEraBlock) -> Result<(), StoreError> {
+        let mut tx_table: Table<TxKeyType, TxValueType> = write_tx
+            .open_table(TX_TABLE)
+            .map_err(|e| StoreError::ReDBError(ReDBError::TableError(e)))?;
+        for tx in block.txs() {
+            tx_table
+                .insert(tx.hash().deref(), tx.encode().as_slice())
+                .map_err(|e| StoreError::ReDBError(ReDBError::InsertError(e)))?;
+            Self::store_utxos(write_tx, &tx)?;
+        }
+        Ok(())
+    }
+
+    fn store_utxos(write_tx: &WriteTransaction, tx: &MultiEraTx) -> Result<(), StoreError> {
+        let mut utxo_table: Table<UTxOKeyType, UTxOValueType> = write_tx
+            .open_table(UTXO_TABLE)
+            .map_err(|e| StoreError::ReDBError(ReDBError::TableError(e)))?;
+        let mut utxo_by_addr_table: MultimapTable<UTxOByAddrKeyType, UTxOByAddrValueType> =
+            write_tx
+                .open_multimap_table(UTXO_BY_ADDR_TABLE)
+                .map_err(|e| StoreError::ReDBError(ReDBError::TableError(e)))?;
+        let mut utxo_by_beacon_table: MultimapTable<UTxOByBeaconKeyType, UTxOByBeaconValueType> =
+            write_tx
+                .open_multimap_table(UTXO_BY_BEACON_TABLE)
+                .map_err(|e| StoreError::ReDBError(ReDBError::TableError(e)))?;
+        for (index, output) in tx.outputs().iter().enumerate() {
+            utxo_table
+                .insert(
+                    (tx.hash().deref().as_slice(), index as u8),
+                    output.encode().as_slice(),
+                )
+                .map_err(|e| StoreError::ReDBError(ReDBError::InsertError(e)))?;
+            let addr: Vec<u8> = output
+                .address()
+                .map_err(StoreError::AddressDecoding)?
+                .to_vec();
+            utxo_by_addr_table
+                .insert(addr.as_slice(), (tx.hash().deref().as_slice(), index as u8))
+                .map_err(|e| StoreError::ReDBError(ReDBError::InsertError(e)))?;
+            for policy in output
+                .non_ada_assets()
+                .iter()
+                .map(MultiEraPolicyAssets::policy)
+            {
+                utxo_by_beacon_table
+                    .insert(policy.deref(), (tx.hash().deref().as_slice(), index as u8))
+                    .map_err(|e| StoreError::ReDBError(ReDBError::InsertError(e)))?;
+            }
+        }
+        Ok(())
```

Comment:
We're missing the update of the data for the consumed utxos. Eg: when an utxo is consumed in a tx, we should remove it from the tables and indices.

Reply:
@MaicoLeberle when processing consumed utxos, we should also remove the corresponding entries from the indices (eg: remove the utxo (hash, index) from the `utxo_by_address`, `utxo_by_beacon`, etc.

you can do this by retrieving the utxo body from the table and re-computing the address, tokens, etc.